include/seqan3/io/sequence/sequence_file_out_options.hpp

```diff
@@ -0,0 +1,64 @@
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::sequence_file_out_options.
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3
+{
+
+//!\brief The options type defines various option members that influence the behaviour of all or some formats.
+struct sequence_file_out_options
+{
+    //!\brief Begin the ID line with ";" instead of ">" (not recommended).
+    bool        fasta_legacy_id_marker  = false;
+    //!\brief Insert a single space after ">" (or ";") before the actual ID.
+    bool        fasta_blank_before_id   = true;
+    //!\brief Inserts linebreaks after every n-th letter in the sequence; 0 means no linebreaks.
+    uint32_t    fasta_letters_per_line  = 80;
+//     bool        fasta_charcounts        = false;
+
+    /*!\brief The default plain text line-ending is "\n", but on Windows an additional carriage return is
+        *        recommended ("\r\n" for line-ending).
+        */
+    bool        add_carriage_return     = false;
```

Comment:
should we ever want this? 
Does it need to be handled as carriage return on Windows. I mean must this be configurable?

Reply:
It is not specified in the format so both are valid. It should not depend on the compiling platform, but on the use-case. Imagine a server that produces files for a lab where everyone uses windows and often looks at FASTA files with an editor (I fear this scenario is quite frequent). You will want the server to produce files with CRs, otherwise the files will be unreadable.

Reply:
But, wouldn't the editor take care of this in general. I mean they won't use vim on Windows. So  gui based editors could probably deal with both endings? Just thinking, if it would be possible to reduce complexity here.